### DISCUSSION TITLE
Fastest way to check if initial commit?

### DISCUSSION THREAD

**Discussion by trapexit** [JacksonAllan/Verstable#14] [Q&A] [answered]
"Determine whether the key was inserted by comparing the table's size before and after the call."

My use case is that I have a stream of c strings which effectively need to dedup. I can only have 1 output regardless of the number of inputs of that particular string. I've been using khash which will indicate in the return value whether or not the insert was new or overwrite. Is the above the best way to do the same with verstable? Is the cost of looking up table size 2x sufficiently small to make that a reasonable pattern?

**Comment by JacksonAllan** [comment, marked as answer]
Hi!

>Is the above the best way to do the same with verstable?

Yes, that's the intended usage. It's a bit clunky, but when designing the API, I thought this approach was better than the alternatives (namely returning a struct or requiring users to pass in pointer to a `bool` even though they might not need this information).

>Is the cost of looking up table size 2x sufficiently small to make that a reasonable pattern?

Yes. Verstable tracks the current number of keys currently in the hash table, so the `_size` function just [returns that value](https://github.com/JacksonAllan/Verstable/blob/099ebd6586b115646ef7b98e0e23de6a9792e01e/verstable.h#L986). The only reasons that there is a function to retrieve this value are, firstly, API consistency and, secondly, to keep the struct members an implementation detail.
